legate/core/communicator.py

```diff
@@ -147,6 +149,12 @@ def _initialize(self, volume: int) -> FutureMap:
             side_effect=True,
         )
         cpucoll_id = task.execute_single()
+        cpucoll_id.wait()
```

Comment:
this is very bad. why do you need to block on the task? this will stall the whole downstream pipeline.